tests/integration_tests/tests/agentless_tests/component/test_component_basics.py

```diff
@@ -76,6 +77,85 @@ def test_component_creation_with_secrets_and_plugins(self):
                           deployment_id)
 
 
+class ComponentPluginsTest(AgentlessTestCase):
+    TEST_PACKAGE_NAME = 'cloudify-script-plugin'
+    TEST_PACKAGE_VERSION = '1.2'
+    test_blueprint = """
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - cloudify/types/types.yaml
+
+node_templates:
+  component_node:
+    type: cloudify.nodes.Component
+    properties:
+      resource_config:
+        blueprint:
+          external_resource: true
+          id: basic
+        deployment:
+          id: component
+      plugins:
+        cloudify-script-plugin:
+          wagon_path: {0}/{1}
+          plugin_yaml_path: {0}/{2}
+"""
+
+    def test_not_loading_existing_plugin(self):
+        mock_id = upload_mock_plugin(self.TEST_PACKAGE_NAME,
+                                     self.TEST_PACKAGE_VERSION)['id']
+        self.wait_for_all_executions_to_end()
+        basic_blueprint_path = resource('dsl/empty_blueprint.yaml')
+        self.client.blueprints.upload(basic_blueprint_path,
+                                      entity_id='basic')
+
+        deployment_id = 'd{0}'.format(uuid.uuid4())
+        main_blueprint = self.test_blueprint.format(
+            'https://cloudify-tests-files.s3-eu-west-1.amazonaws.com/plugins',
+            'cloudify_script_plugin/1_2/'
+            'cloudify_script_plugin-1.2-py27-none-any.wgn',
+            'cloudify_script_plugin/1_2/plugin.yaml'
+        )
+
+        blueprint_path = self.make_yaml_file(main_blueprint)
+        self.deploy_application(blueprint_path, deployment_id=deployment_id)
+        plugins_list = self.client.plugins.list()
+        self.assertTrue(plugins_list[0]['package_name'],
+                        self.TEST_PACKAGE_NAME)
+        self.assertEqual(len(self.client.plugins.list()), 1)
```

Comment:
when asserting len AND asserting something on `plugins_list[0]`, please change the ordering to assert on the length FIRST, so that if the list is empty, we get a test failure, not an IndexError in the `[0]`